numpy/lib/tests/test_shape_base.py

```diff
@@ -289,14 +289,26 @@ def test_functionality(self):
             assert_(b.shape[axis] == 1)
             assert_(np.squeeze(b).shape == s)
 
-    def test_deprecations(self):
-        # 2017-05-17, 1.13.0
+    def test_axis_tuple(self):
+        a = np.empty((3, 3, 3))
+        assert np.expand_dims(a, axis=(0, 1, 2)).shape == (1, 1, 1, 3, 3, 3)
+        assert np.expand_dims(a, axis=(0, -1, -2)).shape == (1, 3, 3, 3, 1, 1)
+        assert np.expand_dims(a, axis=(0, 3, 5)).shape == (1, 3, 3, 1, 3, 1)
+        assert np.expand_dims(a, axis=(0, -3, -5)).shape == (1, 1, 3, 1, 3, 3)
```

Comment:
I would like a test for well defined behaviour if an axis is repeated. Likely, that is simple an additional `len(shape)` check above. Although I may be missing something.

Reply:
If an axis is repeated, a `ValueError` is raised.  There's a test for that here:
https://github.com/numpy/numpy/pull/14051/files#diff-054e272f56fd4778417dcb8731d51476R309

Reply:
Sorry, yeah, just realized that normalize axis tuple does this. Thanks for pointing it out!